SU2_CFD/src/solvers/CSolver.cpp

```diff
@@ -1818,6 +1823,7 @@ void CSolver::CompleteComms(CGeometry *geometry,
 
       SU2_OMP_MASTER
       SU2_MPI::Waitany(geometry->nP2PRecv, geometry->req_P2PRecv, &ind, &status);
+      END_SU2_OMP_MASTER
```

Comment:
Is there a way in OpDiLib to debug missing end pragmas?

Reply:
There is no dedicated way yet, but I am working on it. Most of the macros involve unbalanced `{` or `}` so that improper pairing leads to incorrect code. This can be "debugged" with the compiler. However, there are also more subtle errors like using the wrong end macro. These can be "debugged" by investigating segfaults...

Reply:
OpDiLib offers a syntax checker now. It can also be used with application specific macros (as opposed to the OpDiLib ones) as they are used in SU2. I added a syntax file for this in 6aaebca2b7c3273ef365b57d8352a7339f9f6bfe. It is placed in the root directory for now, we should discuss the location. You can try it out from the root directory via
```
python externals/opdi/syntax/check.py su2omp.syntax.json Common SU2_CFD -p *.hpp *.cpp *.inl -r -s
```